include/zephyr/net/net_mgmt.h

```diff
@@ -76,6 +76,29 @@ struct net_if;
 
 /** @endcond */
 
+/** Central place the definition of the layer codes (7 bit value) */
```

Comment:
Defining these layer codes as integers rather than individual bits essentially makes the `global_event_mask` optimisation in `net_mgmt.c` pointless. This is no different from the current behaviour, but worth looking at if we are redoing the definitions of events in order to get it right the first time?

Reply:
You mentioned earlier the problems with `net_mgmt_event_wait()` but it seems that I missed your point there as after looking the API it looks ok after all. Or were you just wanting to avoid checking enum values in the first place and just use bit values?